client/contract.go

```diff
@@ -24,37 +24,42 @@ func NewContractClient(c ethclient.Client) *ContractClient {
 	}
 }
 
-// GetContractCurrency returns the currency for a specific address
-func (c *ContractClient) GetContractCurrency(addr common.Address, erc20 bool) (*ContractCurrency, error) {
+// GetContractInfo returns the symbol and decimals for [addr].
+func (c *ContractClient) GetContractInfo(addr common.Address, erc20 bool) (string, uint8, error) {
+	// We don't define another struct because this is never used outside of this
+	// function.
+	type ContractInfo struct {
+		Symbol   string
+		Decimals uint8
+	}
+
 	if currency, cached := c.cache.Get(addr); cached {
-		return currency.(*ContractCurrency), nil
+		cast := currency.(*ContractInfo)
+		return cast.Symbol, cast.Decimals, nil
 	}
 
 	token, err := NewContractInfoToken(addr, c.ethClient)
 	if err != nil {
-		return nil, err
+		return "", 0, err
 	}
 
-	symbol, symbolErr := token.Symbol(nil)
-	decimals, decimalErr := token.Decimals(nil)
-
-	// Any of these indicate a failure to get complete information from contract
-	if symbolErr != nil || decimalErr != nil || symbol == "" || decimals == 0 {
+	// [symbol] is set to "" if [token.Symbol] errors.
+	symbol, _ := token.Symbol(nil)
+	if symbol == "" {
 		if erc20 {
 			symbol = UnknownERC20Symbol
-			decimals = UnknownERC20Decimals
 		} else {
 			symbol = UnknownERC721Symbol
-			decimals = UnknownERC721Decimals
 		}
 	}
 
-	currency := &ContractCurrency{
-		Symbol:   symbol,
-		Decimals: int32(decimals),
-	}
+	// [decimals] is set to 0 if [token.Decimals] errors.
+	decimals, _ := token.Decimals(nil)
```

Comment:
There can also be tokens with no decimals though? In that case, you are silencing the errror.

Reply:
The existing code only returns an error if `token` fails to initialize: https://github.com/ava-labs/avalanche-rosetta/blob/master/client/contract.go#L33-L36

if `token.Decimals()` ever errors, we default to 0: https://github.com/ava-labs/avalanche-rosetta/blob/master/client/types.go#L11-L15

Since it's already set to 0 on error (via `*new(uint8)`), we don't need to check this error: https://github.com/ava-labs/avalanche-rosetta/blob/master/client/contractInfoToken.go#L185-L197